### PR TITLE
Update README.md to provide installation guide for mac / linux users

### LINKED ISSUE #320: Installing the crate in a macbook machine
I tried to install this crate into my macbook machine following the documentation. So I run: 

`cargo install --locked cargo-spellcheck`

which returned the following error when compiling:

```
error: failed to run custom build command for `hunspell-sys v0.3.1`

Caused by:
  process didn't exit successfully: `/var/folders/0n/fkgm548x15q996h51xlpc2rh0000gn/T/cargo-installS4VX2v/release/build/hunspell-sys-2aba3f0e9baf644a/build-script-build` (signal: 6, SIGABRT: process abort signal)
  --- stderr
  dyld[28610]: Library not loaded: @rpath/libclang.dylib
    Referenced from: <4E7CEE65-E915-386C-83A4-83815EA2231B> /private/var/folders/0n/fkgm548x15q996h51xlpc2rh0000gn/T/cargo-installS4VX2v/release/build/hunspell-sys-2aba3f0e9baf644a/build-script-build
    Reason: tried: '/var/folders/0n/fkgm548x15q996h51xlpc2rh0000gn/T/cargo-installS4VX2v/release/deps/libclang.dylib' (no such file), '/var/folders/0n/fkgm548x15q996h51xlpc2rh0000gn/T/cargo-installS4VX2v/release/libclang.dylib' (no such file)
```

It's complaining that the dynamic linker is unable to locate the libclang.dylib library on the machine. After some googling, I managed to resolve the issue with the following command

`export DYLD_FALLBACK_LIBRARY_PATH="$(xcode-select --print-path)/Toolchains/XcodeDefault.xctoolchain/usr/lib/" && cargo install --locked cargo-spellcheck`

I believe we can enrich the installation guide a bit to specify the necessary env variable that needs to be present in order to install the crate successfully (for all the 3 OS: Windows, Linux & Mac).

### PR DESCRIPTION
**Pull request by Owen-CH-Leung**
* 📙 Documentation : Update the `README.md` to provide a detailed installation guide for mac & linux users.

Closes #320  .

## 📜 Checklist

 * [X] Works on the `./demo` sub directory
 * [X] Test coverage is excellent and passes
 * [X] Documentation is thorough

### CODE REVIEW

**Comment by Owen-CH-Leung** [comment]
@drahnr Just fixed the failing CI test. Can I seek your review ? Many thanks

File: README.md [hunk 1]
```diff
@@ -61,6 +61,33 @@ cargo spellcheck fix
 The `--locked` flag is the preferred way of installing to get the tested set of
 dependencies.
 
+in Mac, you'd need to ensure that `libclang.dylib` can be found by the linker
```
Comment (drahnr) [inline README.md:64]:
```suggestion
on OS X, you need to ensure that `libclang.dylib` can be found by the linker
```

File: README.md [hunk 2]
```diff
@@ -61,6 +61,33 @@ cargo spellcheck fix
...
+
+This can be achieved by setting the environment variable:
+
+```
+DYLD_FALLBACK_LIBRARY_PATH
+```
```
Comment (drahnr) [inline README.md:70]:
```suggestion 
```

File: README.md [hunk 3]
```diff
@@ -61,6 +61,33 @@ cargo spellcheck fix
...
 
+in Mac, you'd need to ensure that `libclang.dylib` can be found by the linker
+
+This can be achieved by setting the environment variable:
+
+```
+DYLD_FALLBACK_LIBRARY_PATH
+``` 
+
+Run the following command before installing the crate:
```
Comment (drahnr) [inline README.md:72]:
```suggestion
which can be achieved by setting `DYLB_FALLBACK_LIBRARY_PATH`:
```

**Review by drahnr — APPROVED** [review]
A few nits, otherwise LGTM. Thank you! Sorry for the long delay, life happened

**Comment by drahnr** [comment]
Could you squash it into one commit, then we can merge it :) Thank you!

**Comment by Owen-CH-Leung** [comment]
@drahnr Here: 

https://github.com/drahnr/cargo-spellcheck/pull/331

**Pull request closed without merging**